libpod/sqlite_state.go

```diff
@@ -0,0 +1,2535 @@
+// ID. Returns ErrNoSuchCtr if the ID does not exist.
+func (s *SQLiteState) GetContainerName(id string) (string, error) {
+	if id == "" {
+		return "", define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return "", define.ErrDBClosed
+	}
+
+	var name string
+
+	row := s.conn.QueryRow("SELECT Name FROM ContainerConfig WHERE ID=?;", id)
+	if err := row.Scan(&name); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return "", define.ErrNoSuchCtr
+		}
+
+		return "", fmt.Errorf("looking up container %s name: %w", id, err)
+	}
+
+	return name, nil
+}
+
+// GetPodName returns the name of the pod associated with a given ID.
+// Returns ErrNoSuchPod if the ID does not exist.
+func (s *SQLiteState) GetPodName(id string) (string, error) {
+	if id == "" {
+		return "", define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return "", define.ErrDBClosed
+	}
+
+	var name string
+
+	row := s.conn.QueryRow("SELECT Name FROM PodConfig WHERE ID=?;", id)
+	if err := row.Scan(&name); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return "", define.ErrNoSuchPod
+		}
+
+		return "", fmt.Errorf("looking up pod %s name: %w", id, err)
+	}
+
+	return name, nil
+}
+
+// Container retrieves a single container from the state by its full ID
+func (s *SQLiteState) Container(id string) (*Container, error) {
+	if id == "" {
+		return nil, define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return nil, define.ErrDBClosed
+	}
+
+	ctrConfig, err := s.getCtrConfig(id)
+	if err != nil {
+		return nil, err
+	}
+
+	ctr := new(Container)
+	ctr.config = ctrConfig
+	ctr.state = new(ContainerState)
+	ctr.runtime = s.runtime
+
+	if err := finalizeCtrSqlite(ctr); err != nil {
+		return nil, err
+	}
+
+	return ctr, nil
+}
+
+// LookupContainerID retrieves a container ID from the state by full or unique
+// partial ID or name
+func (s *SQLiteState) LookupContainerID(idOrName string) (string, error) {
+	if idOrName == "" {
+		return "", define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return "", define.ErrDBClosed
+	}
+
+	rows, err := s.conn.Query("SELECT ID FROM ContainerConfig WHERE ContainerConfig.Name=? OR (ContainerConfig.ID LIKE ?);", idOrName, idOrName)
+	if err != nil {
+		return "", fmt.Errorf("looking up container %q in database: %w", idOrName, err)
+	}
+	defer rows.Close()
+
+	var id string
+	foundResult := false
+	for rows.Next() {
+		if foundResult {
+			return "", fmt.Errorf("more than one result for container %q: %w", idOrName, define.ErrCtrExists)
+		}
+
+		if err := rows.Scan(&id); err != nil {
+			return "", fmt.Errorf("retrieving container %q ID from database: %w", idOrName, err)
+		}
+		foundResult = true
+	}
+	if !foundResult {
+		return "", define.ErrNoSuchCtr
+	}
+
+	return id, nil
+}
+
+// LookupContainer retrieves a container from the state by full or unique
+// partial ID or name
+func (s *SQLiteState) LookupContainer(idOrName string) (*Container, error) {
+	if idOrName == "" {
+		return nil, define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return nil, define.ErrDBClosed
+	}
+
```

Comment:
can this use Lookupcontainerorid? Or perhaps an unexported func that prevents the duplicate query between funcs?

Reply:
Duplication saves us one query, so it's a bit faster this was, but we can probably share parts of the logic.